### DISCUSSION TITLE
customizing for start

### DISCUSSION THREAD

**Discussion by stiebel** [xournalpp/xournalpp#3898] [Ideas]
Hi,
I want to use xournal as a standard whiteboard app in our linux system at our school.
So I want to have a customized start-behavior:

start in fullscreen
have customized toolbars 
have a differnet standard background

And than I would like to have extended options just for whiteboards like:
tollbar at the bottom instead of top,
automatic save as pdf
-> if switched on, do not ask for saving "normal" file at the end
landscape (as default) for pages.

Is there a chance to have this features allready (and I did not find them)
or to get them somehow?

Greetings,
Roland

**Comment by rolandlo** [comment]
Hi @stiebel 

Yes, you can do most of that already:
- have a customized toolbar (at the bottom): Just use menu `View > Toolbar > Customize` to customize your toolbar. Place all your icons at the bottom, if you want them to be there.
- have a different standard background (with landscape per default): Just use menu `Journal > Customize Page Template`
- start in fullscreen mode: Assuming you run the X11 window manager and have `xdotool` installed, start Xournal++ via  
 ```
xournalpp & sleep 1s; xdotool key F11
```
You could write a bash script with that content that you could run from the file manager or from the dock.

Now about automatic save as pdf: What exactly do you want? Periodic save as pdf (say every minute) or just save as pdf when you would otherwise save the xopp-file?

**Comment by stiebel** [comment]
Hi Rolandlo,
thanks for your answer.
Great, xournal++ is much more customizable than I thought.
I managed to make all the changes I wanted.

However, we will use xournal++ as our standard whiteboard software, but our students have to get the board-files readable without special software, so we have to "export" everything to pdf when ready.
What I would love to see is a feature like the autosave that keeps saving the actual document as a pdf file every minute and when closing without asking. When we mount our cloud/share/folder to a place to save the documents to, we would have an automatic "board distributor" for all teachers without them having to understand whats going on.

This looks to me like an unusual but still possible wish.

Greetings
Roland

**Comment by rolandlo** [comment]
Exporting to pdf once every minute could easily be achieved via a [plugin](https://xournalpp.github.io/guide/plugins/plugins/). The `app.export` API needed is not yet documented on the website, but you get the info from PR https://github.com/xournalpp/xournalpp/pull/3566. Exporting at the end and closing without asking to save the .xopp file is not supported from the plugin API though. 
See also PR https://github.com/xournalpp/xournalpp/pull/2618 and PR https://github.com/xournalpp/xournalpp/pull/3700
